### ISSUE TITLE
Loss of Named stars and NGC in result

### ISSUE THREAD

**Issue by AstroKeith** [dstndstn/astrometry.net#288]
I have just migrated from 0.90 to 0.93 (note downloading 'latest' gets 0.93, not 0.94)

Previously in a solve result, at the end, was a short list of bright stars and ngc objects found.

I'm not getting this any more. Any way to get it back please. In particular I just need the name of the brightest named star in the field.

I'm using same options as with 0.90  ..... 
--overwrite --skip-solved --cpulimit 10 --downsample 2 --no-remove-lines --uniformize 0 --scale-units arcsecperpix --scale-low 14.9 --scale-high 15.1 --new-fits none --solved none --match none --corr none --rdls none

TIA
Keith

**Comment by dstndstn** [comment]
Hi,
I just reviewed the changes between 0.90 and 0.93 and I really don't see anything that would cause this.  Would you be able to generate log files (add a "--verbose" for more details) for 0.90 and 0.93?
I believe the Debian packages removed some object annotations (NGC/IC?) because of licensing concerns, but it sounds like you're building from source rather than using a package.

**Comment by dstndstn** [comment]
Or, alternatively -- there aren't any major changes from 0.90 to 0.93, so unless you need something new, just keep using 0.90!

**Comment by AstroKeith** [comment]
I'll try to get those log files. yes I'm building from source.
The update to 0.93 was driven by the new Raspberry Pi 5 and its 64bit Bookworm OS. It comes with Numpy 1.24 bundled in so I thought I'd go the rest of the way.
BTW the Pi5 with Bookworm runs astrometry.net very nicely - I'm getting blind solves in 0.7 seconds for a 1 second exposure image. This is x3 faster than the Pi4. The annotation runs even quicker, about x6 faster.

**Comment by AstroKeith** [comment]
Below are the two results for the version 0.94 and 0.90. I can see that with 0.90 the solver goes on to run some plot binaries, which doesnt happen on 0.94. The Python script is identical for both with the options I reported earlier.
ver 0.94 is running on a Pi5 under 64bit Bookworm OS, with Numpy 1.24.1 and Python 11.1
ver 0.90 is running on a Pi4 under 32bit Bullseye OS, Numpy 1.22.0 and Python 3.9 

I do remember seeing this issue on the Pi4 a while back when I first tried 0.93 (32bit Bullseye OS, Numpy 1.22.0 and Python 3.9)

First here is the ver 0.94 result:
Running solver:
solver run parameters:
indexes:
  /usr/local/astrometry/data/index-4115.fits
  /usr/local/astrometry/data/index-4114.fits
  /usr/local/astrometry/data/index-4113.fits
  /usr/local/astrometry/data/index-4112.fits
  /usr/local/astrometry/data/index-4111.fits
  /usr/local/astrometry/data/index-4110.fits
  /usr/local/astrometry/data/index-4109.fits
  /usr/local/astrometry/data/index-4108.fits
  /usr/local/astrometry/data/index-4107.fits
fieldfname /var/tmp/solve/capture.axy
fields 1 
fieldid 0
matchfname /tmp/tmp.match.zgbLtL
wcs /var/tmp/solve/capture.wcs
fieldid_key FIELDID
indexrdlsfname /tmp/tmp.rdls.uUtVf9
parity 2
codetol 0.01
startdepth 190
enddepth 200
fieldunits_lower 13.86
fieldunits_upper 16.94
verify_pix 1
xcolname X
ycolname Y
maxquads 0
maxmatches 0
cpulimit 5.000000
timelimit 0
total_timelimit 0
total_cpulimit 5.000000
Reading fields file /var/tmp/solve/capture.axy...found 1 fields.
cx<=dx constraints: 0
meanx constraints: 135
RA,Dec constraints: 0
AB scale constraints: 0
Spent 0.175242 seconds on this field.
Field: /var/tmp/solve/capture.jpg
Field center: (RA,Dec) = (38.609413, 88.942447) deg.
Field center: (RA H:M:S, Dec D:M:S) = (02:34:26.259, +88:56:32.808).
Field size: 5.46839 x 4.1042 degrees
Field rotation angle: up is -147.06 degrees E of N
Field parity: neg
Deleting temp file /tmp/tmp.match.zgbLtL
Deleting temp file /tmp/tmp.rdls.uUtVf9
Deleting temp file /tmp/tmp.ppm.fBYbde

Now the ver 0.90
Running solver:
solver run parameters:
indexes:
  /usr/local/astrometry/data/index-4113.fits
  /usr/local/astrometry/data/index-4112.fits
  /usr/local/astrometry/data/index-4111.fits
  /usr/local/astrometry/data/index-4110.fits
fieldfname /var/tmp/solve/capture.axy
fields 1 
fieldid 0
matchfname /tmp/tmp.match.kUfwHF
wcs /var/tmp/solve/capture.wcs
fieldid_key FIELDID
indexrdlsfname /tmp/tmp.rdls.BtOXpG
parity 2
codetol 0.01
startdepth 190
enddepth 200
fieldunits_lower 13.5
fieldunits_upper 16.5
verify_pix 1
xcolname X
ycolname Y
maxquads 0
maxmatches 0
cpulimit 5.000000
timelimit 0
total_timelimit 0
total_cpulimit 5.000000
Reading fields file /var/tmp/solve/capture.axy...found 1 fields.
cx<=dx constraints: 0
meanx constraints: 37
RA,Dec constraints: 0
AB scale constraints: 0
Spent 0.460383 seconds on this field.
Field: /var/tmp/solve/capture.jpg
Field center: (RA,Dec) = (38.609413, 88.942447) deg.
Field center: (RA H:M:S, Dec D:M:S) = (02:34:26.259, +88:56:32.808).
Field size: 5.46839 x 4.1042 degrees
Field rotation angle: up is -147.06 degrees E of N
Field parity: neg
Creating index object overlay plot...
Running:
  /usr/local/astrometry/bin/plotxy -I /tmp/tmp.ppm.t8i8CF -i /var/tmp/solve/capture.axy -C red -w 2 -r 6 -N 200 -x 1 -y 1 -P | /usr/local/astrometry/bin/plotxy -i /var/tmp/solve/capture-indx.xyls -I - -w 2 -r 4 -C green -x 1 -y 1  -P | /usr/local/astrometry/bin/plotquad -I - -C green -w 2 -d 3  648.703  25.6766  819.424  941.604  696.736  431.262 > /var/tmp/solve/capture-indx.png
Running: /usr/local/astrometry/bin/plotxy -I /tmp/tmp.ppm.t8i8CF -i /var/tmp/solve/capture.axy -C red -w 2 -r 6 -N 200 -x 1 -y 1 -P | /usr/local/astrometry/bin/plotxy -i /var/tmp/solve/capture-indx.xyls -I - -w 2 -r 4 -C green -x 1 -y 1  -P | /usr/local/astrometry/bin/plotquad -I - -C green -w 2 -d 3  648.703  25.6766  819.424  941.604  696.736  431.262 > /var/tmp/solve/capture-indx.png
Creating annotation plot...
Running:
  /usr/local/astrometry/bin/plot-constellations -v -w /var/tmp/solve/capture.wcs -i /tmp/tmp.ppm.t8i8CF -N -B -C -o /var/tmp/solve/capture-ngc.png
Your field contains:
  The star Polaris / Alrucaba / Cinosura / Tramontana / Yilduz / Mismar (Î± UMi / 1 UMi)

**Comment by dstndstn** [comment]
Okay, great.  My guess is either that the plotting programs didn't get built for some reason, or that the first one crashed.

Can you please send a log from building 0.94 (including "make extra").

And also the full log from running 0.94.

Thanks!

**Comment by AstroKeith** [comment]
Will do. BTW how do I get a full log when running please

**Comment by AstroKeith** [comment]
Here are log files.
I remember during the build that when trying to install libnetpbm10-dev it loaded libnetpbm11-dev instead.
[runfile.txt](https://github.com/dstndstn/astrometry.net/files/13457419/runfile.txt)
[makeextrafile.txt](https://github.com/dstndstn/astrometry.net/files/13457420/makeextrafile.txt)
[makefile.txt](https://github.com/dstndstn/astrometry.net/files/13457421/makefile.txt)
[makeinstallfile.txt](https://github.com/dstndstn/astrometry.net/files/13457422/makeinstallfile.txt)
[makepyfile.txt](https://github.com/dstndstn/astrometry.net/files/13457423/makepyfile.txt)

**Comment by AstroKeith** [comment]
BTW  I can run plotann.py and get successful graphics overlay

Thanks
Keith

**Comment by AstroKeith** [comment]
Also, it seems this may be related to issue #276?

**Comment by dstndstn** [comment]
Oh yeah!, that looks like exactly the same issue.  In your logs, I see:
```
os-features-test.c:8:10: fatal error: netpbm/pam.h: No such file or directory
    8 | #include <netpbm/pam.h>
      |          ^~~~~~~~~~~~~~
```
so then it disables reading JPEG, PNG, etc files for annotations.

Can you please try the fix described in that ticket?

(There are two separate annotation programs, plotann.py which is newer, and plot-constellations, which is what solve-field uses.  So being able to run plotann.py isn't enough.)

**Comment by AstroKeith** [comment]
So....
That fix didn't work, but exposed I had a bad install of netpbm. I un-installed netpbm and libnetpbm11-dev, rebuilt astrometry 0.94 and so far all is OK. Great! (I didnt need the symlink)
Some thoughts.....
When I first installed libnetpbm10-dev I was told the loader had selected libnetpbm11-dev instead. Maybe that interrupted the subsequent netpbm install? On re-install I went straight for libnetpbm11-dev.
I tried just rebuilding 0.94 but no change. I had to delete and start with a fresh download.

Thanks for all the help as usual!
Keith
